### PR TITLE
Migrate multiple tests from Boost to Catch2

### PR DESCRIPTION
**Pull request by Aakash-kaushik**
List of migrated tests: Issue #2523 
- drusilla_select_test.cpp
- emst_test.cpp
- fastmks_test.cpp
- facilities_test.cpp
- hpt_test.cpp
- hyperplane_test.cpp
- init_rules_test.cpp
- linear_svm_test.cpp
- main_tests/emst_test.cpp
- main_tests/fastmks_test.cpp
- main_tests/linear_svm_test.cpp

### CODE REVIEW

File: src/mlpack/tests/drusilla_select_test.cpp [hunk 1]
```diff
@@ -64,20 +62,20 @@ BOOST_AUTO_TEST_CASE(DrusillaSelectExhaustiveExactTest)
...
+  REQUIRE(neighborsTrue.n_rows == neighbors.n_rows);
+  REQUIRE(distancesTrue.n_cols == distances.n_cols);
+  REQUIRE(distancesTrue.n_rows == distances.n_rows);
 
   for (size_t i = 0; i < distances.n_elem; ++i)
   {
-    BOOST_REQUIRE_EQUAL(neighbors[i], neighborsTrue[i]);
-    BOOST_REQUIRE_CLOSE(distances[i], distancesTrue[i], 1e-5);
+    REQUIRE(neighbors[i] == neighborsTrue[i]);
+    REQUIRE(distances[i] == Approx(distancesTrue[i]).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/drusilla_select_test.cpp:73]:
```suggestion
    REQUIRE(distances[i] == Approx(distancesTrue[i]).epsilon(1e-7));
```

File: src/mlpack/tests/drusilla_select_test.cpp [hunk 2]
```diff
@@ -122,34 +120,34 @@ BOOST_AUTO_TEST_CASE(SerializationTest)
...
-    BOOST_REQUIRE_EQUAL(neighbors[i], neighborsText[i]);
-    BOOST_REQUIRE_EQUAL(neighbors[i], neighborsBinary[i]);
+    REQUIRE(neighbors[i] == neighborsXml[i]);
+    REQUIRE(neighbors[i] == neighborsText[i]);
+    REQUIRE(neighbors[i] == neighborsBinary[i]);
 
-    BOOST_REQUIRE_CLOSE(distances[i], distancesXml[i], 1e-5);
-    BOOST_REQUIRE_CLOSE(distances[i], distancesText[i], 1e-5);
-    BOOST_REQUIRE_CLOSE(distances[i], distancesBinary[i], 1e-5);
+    REQUIRE(distances[i] == Approx(distancesXml[i]).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/drusilla_select_test.cpp:143]:
```suggestion
    REQUIRE(distances[i] == Approx(distancesXml[i]).epsilon(1e-7));
```

Same for others too

File: src/mlpack/tests/emst_test.cpp [hunk 3]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 0), newFromOld[1]);
-    BOOST_REQUIRE_EQUAL(results(0, 0), newFromOld[8]);
+    REQUIRE(results(1, 0) == newFromOld[1]);
+    REQUIRE(results(0, 0) == newFromOld[8]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 0), 0.08, 1e-5);
+  REQUIRE(results(2, 0) == Approx(0.08).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:70]:
```suggestion
  REQUIRE(results(2, 0) == Approx(0.08).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 4]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 1), newFromOld[9]);
-    BOOST_REQUIRE_EQUAL(results(0, 1), newFromOld[10]);
+    REQUIRE(results(1, 1) == newFromOld[9]);
+    REQUIRE(results(0, 1) == newFromOld[10]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 1), 0.09, 1e-5);
+  REQUIRE(results(2, 1) == Approx(0.09).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:82]:
```suggestion
  REQUIRE(results(2, 1) == Approx(0.09).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 5]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 2), newFromOld[0]);
-    BOOST_REQUIRE_EQUAL(results(0, 2), newFromOld[2]);
+    REQUIRE(results(1, 2) == newFromOld[0]);
+    REQUIRE(results(0, 2) == newFromOld[2]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 2), 0.1, 1e-5);
+  REQUIRE(results(2, 2) == Approx(0.1).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:94]:
```suggestion
  REQUIRE(results(2, 2) == Approx(0.1).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 6]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 3), newFromOld[1]);
-    BOOST_REQUIRE_EQUAL(results(0, 3), newFromOld[2]);
+    REQUIRE(results(1, 3) == newFromOld[1]);
+    REQUIRE(results(0, 3) == newFromOld[2]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 3), 0.22, 1e-5);
+  REQUIRE(results(2, 3) == Approx(0.22).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:106]:
```suggestion
  REQUIRE(results(2, 3) == Approx(0.22).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 7]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 4), newFromOld[3]);
-    BOOST_REQUIRE_EQUAL(results(0, 4), newFromOld[10]);
+    REQUIRE(results(1, 4) == newFromOld[3]);
+    REQUIRE(results(0, 4) == newFromOld[10]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 4), 0.25, 1e-5);
+  REQUIRE(results(2, 4) == Approx(0.25).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:118]:
```suggestion
  REQUIRE(results(2, 4) == Approx(0.25).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 8]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 5), newFromOld[0]);
-    BOOST_REQUIRE_EQUAL(results(0, 5), newFromOld[5]);
+    REQUIRE(results(1, 5) == newFromOld[0]);
+    REQUIRE(results(0, 5) == newFromOld[5]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 5), 0.27, 1e-5);
+  REQUIRE(results(2, 5) == Approx(0.27).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:130]:
```suggestion
  REQUIRE(results(2, 5) == Approx(0.27).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 9]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 6), newFromOld[8]);
-    BOOST_REQUIRE_EQUAL(results(0, 6), newFromOld[9]);
+    REQUIRE(results(1, 6) == newFromOld[8]);
+    REQUIRE(results(0, 6) == newFromOld[9]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 6), 0.46, 1e-5);
+  REQUIRE(results(2, 6) == Approx(0.46).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:142]:
```suggestion
  REQUIRE(results(2, 6) == Approx(0.46).epsilon(1e-7));
```

File: src/mlpack/tests/emst_test.cpp [hunk 10]
```diff
@@ -61,138 +59,138 @@ BOOST_AUTO_TEST_CASE(ExhaustiveSyntheticTest)
...
   }
   else
   {
-    BOOST_REQUIRE_EQUAL(results(1, 7), newFromOld[6]);
-    BOOST_REQUIRE_EQUAL(results(0, 7), newFromOld[7]);
+    REQUIRE(results(1, 7) == newFromOld[6]);
+    REQUIRE(results(0, 7) == newFromOld[7]);
   }
-  BOOST_REQUIRE_CLOSE(results(2, 7), 0.7, 1e-5);
+  REQUIRE(results(2, 7) == Approx(0.7).epsilon(1e-5));
```
Comment (jeffin143) [inline src/mlpack/tests/emst_test.cpp:154]:
```suggestion
  REQUIRE(results(2, 7) == Approx(0.7).epsilon(1e-7));
```

**Review by jeffin143 — CHANGES_REQUESTED** [review]
Took a pass , there are many instances were 
Close when migrated should have the value / 100
Please change it at every location

I was reviewing it through mobile :) so pointed out some of those

**Comment by Aakash-kaushik** [comment]
> Took a pass , there are many instances were 
> Close when migrated should have the value / 100
> Please change it at every location
> 
> I was reviewing it through mobile :) so pointed out some of those

Hey @jeffin143 will make changes to all the files in morning. 🌅

**Comment by Aakash-kaushik** [comment]
> Took a pass , there are many instances were
> Close when migrated should have the value / 100
> Please change it at every location
> 
> I was reviewing it through mobile :) so pointed out some of those

Got everyone :D

**Comment by Aakash-kaushik** [comment]
```
/home/vsts/work/1/s/src/mlpack/tests/linear_svm_test.cpp:849: FAILED:
  REQUIRE( testAcc == Approx(1.0).epsilon(0.0035) )
with expansion:
  0.996 == Approx( 1.0 )
```
I have increase the tolerance value to 0.0045, because if left 0.0035 it sometimes passes and sometimes does not and will be a hassle in all future PRs.

**Comment by Aakash-kaushik** [comment]
closing in favor of #2678

**Pull request closed without merging**